### ISSUE TITLE
Chapter 2: Evaluate your System

### ISSUE THREAD

**Issue by giovanni5** [ageron/handson-ml#380]
Hi, in the input [107] I don't understand why we have to use transform() and not fit_transform() on the full_pipeline. The output is different, but I can't tell why.

**Comment by qiaozhi9092** [comment]
Hi giovanni5,

I can try to answer your question.

The reason Aurélien uses the transform() instead of fit_transform() is because he wants to use the test set to test the performance of the full_pipeline() (which was trained by using the training set). If he used fit_transform() instead, that would defeat the whole propose of using the test set to evaluate the performance of the model, because the full_pipeline would fit the test set and it would give you an illusional low rmse. 

Please let me know if I explained this clearly, or if there is any mistakes, thanks!

**Comment by ageron** [comment]
Thanks for the question @giovanni5 , and for the answer @qiaozhi9092 ! 👍

**Comment by giovanni5** [comment]
It was because the imputer was fitted on the train set and I don't want it to be refitted on the test set?

**Comment by ageron** [comment]
Exactly @giovanni5

**Comment by giovanni5** [comment]
I get it, thank you.
